File: sdks/python/src/bigrag/resources/query.py

from __future__ import annotations

from typing import TYPE_CHECKING
from urllib.parse import quote

from bigrag.types.query import (
    BatchQueryBody,
    BatchQueryResponse,
    MultiQueryBody,
    MultiQueryResponse,
    QueryBody,
    QueryResponse,
)

if TYPE_CHECKING:
    from bigrag._core import BigRAGCore


class QueryResource:

    def __init__(self, client: BigRAGCore) -> None:
        self._client = client

    async def query(self, collection: str, body: QueryBody) -> QueryResponse:
        return await self._client._request(
            "POST",
            f"/v1/collections/{quote(collection, safe='')}/query",
            json=body,
        )

    async def multi_query(self, body: MultiQueryBody) -> MultiQueryResponse:
        return await self._client._request("POST", "/v1/query", json=body)

    async def batch_query(self, body: BatchQueryBody) -> BatchQueryResponse:
        return await self._client._request("POST", "/v1/batch/query", json=body)

File: sdks/typescript/src/resources/query.ts
import type { RequestClient } from "../core.js";
import type {
  BatchQueryBody,
  BatchQueryResponse,
  MultiQueryBody,
  MultiQueryResponse,
  QueryBody,
  QueryResponse,
} from "../types.js";

export class QueryResource {
  constructor(private readonly _client: RequestClient) {}

  query(collection: string, body: QueryBody): Promise<QueryResponse> {
    return this._client._request(
      "POST",
      `/v1/collections/${encodeURIComponent(collection)}/query`,
      { json: body },
    );
  }

  multiQuery(body: MultiQueryBody): Promise<MultiQueryResponse> {
    return this._client._request("POST", "/v1/query", { json: body });
  }

  batchQuery(body: BatchQueryBody): Promise<BatchQueryResponse> {
    return this._client._request("POST", "/v1/batch/query", { json: body });
  }
}
